full_history_avd/src/history_tree/mod.rs

```diff
@@ -176,14 +180,17 @@ impl<SSAVD: SingleStepAVD, HTParams: MerkleTreeParameters> SingleStepAVDWithHist
     pub fn new<R: Rng>(
         rng: &mut R,
         ssavd_pp: &SSAVD::PublicParameters,
-        history_tree_parameters: &<HTParams::H as FixedLengthCRH>::Parameters,
+        history_tree_parameters: &<<<HTStorer as Storer>::P as MerkleTreeParameters>::H as FixedLengthCRH>::Parameters,
     ) -> Result<Self, Error>{
         let ssavd = SSAVD::new(rng, ssavd_pp)?;
-        let history_tree = HistoryTree::new(history_tree_parameters)?;
-        let digest = hash_to_final_digest::<SSAVD, HTParams::H>(
+        // TODO(z-tech): I think this is right from the comments but not sure:
+        let store = HTStorer::new(ssavd_pp, history_tree_parameters).unwrap();
```

Comment:
@nirvantyagi Pretty sure first arg here I need `&[u8]`

Is there a way to easily go from `&SSAVD::PublicParameters` to `&[u8]`?

Reply:
It looks like the ``&[u8]`` that you are expecting is the initial leaf value for the sparse Merkle Tree. The HistoryTree structure builds its sparse Merkle tree using a "default" initial leaf value. You can see how the ``&[u8]`` is generated here: https://github.com/nirvantyagi/vpki/blob/master/full_history_avd/src/history_tree/mod.rs#L30  (``&<[u8; 32]>::default()``)

Seems like ``HTStorer`` might be a misnomer here? This is the SMTStore for the HT? Presumably HT will need a store that also includes the HashMap it maintains.

Reply:
Relatedly, should/will the ``HTStorer`` object also hold the ``SMTStorer`` object within it? Analogous to how HT stores SMT within it. Then this logic of creating the ``SMTStorer`` can fall inside the logic of creating a new ``HTStorer``.